DataDog/integrations-core
# [marathon] adding queue metrics support

### What does this PR do?

This PR attempts to add a few metrics regarding the state of the marathon queues. It also refactors the code originally in the check for a bit of a better structure.

### Motivation

Customer request.

### Testing Guidelines

Updated queue.json with latest v2 JSON provided by Marathon docs (see: https://mesosphere.github.io/marathon/api-console/index.html).  Added (limited) test for generated marathon metrics.

We introduce a couple of tag spaces for two of the metrics (namely: `marathon.queue.offers.reject.last` and `marathon.queue.offers.reject.launch`):
- reason:* (where * is one of `UnfulfilledRole`, `UnfulfilledConstraint`, `NoCorrespondingReservationFound`, `InsufficientCpus`, `InsufficientMemory`, `InsufficientDisk`, `InsufficientGpus`)
- status:{processed, declined}

### Versioning


### Additional Notes

Queue metric documentation: https://github.com/mesosphere/marathon/blob/master/docs/docs/rest-api/public/api/v2/types/queue.raml

JSON examples (also see fixtures):
https://insight.io/github.com/mesosphere/marathon/blob/HEAD/docs/docs/rest-api/public/api/v2/examples/queue.json
